### DISCUSSION TITLE
Understand the code

### DISCUSSION THREAD

**Discussion by TaMaN2031A** [tarantool/tarantool#6906] [Q&A] [answered]
Could you tell me how to have a deep understand of the code? I don't know where to start.

**Comment by Totktonada** [comment, marked as answer]
There is no way to receive a deep understanding of code without a long-lasting working on it.

As for me, the best start is to choose one of tasks, which is marked as a [good first issue](https://github.com/tarantool/tarantool/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22), spot the relevant piece of code (look around `src/` directory content, grep relevant keywords), read it and experiment on it.

There is some internal documentation here: https://docs.tarantool.dev/en/latest/, but not much. I would also suggest to look around articles in Kostja's blog: http://kostja.github.io/ — those are not about particular places in the code, but rather about the architecture and design decisions behind it.

Last, but not least, you can ask a question in the Telegram chat https://t.me/tarantool, but, please, attempt to shrink the question to a certain context (say, particular issue).

*(If you want to participate in GSoC 2022, there is more related Telegram chat, where your questions and answers to them may also be useful for other students: https://t.me/+T7iUBVmNd3sxNjcy. I suggest to use it for all questions more or less related to GSoC.)*
